packages/aws-cdk/lib/api/util/cloudformation/stack-activity-monitor.ts

```diff
@@ -249,6 +250,13 @@ export class StackActivityMonitor {
             event: event,
             metadata: this.findMetadataFor(event.LogicalResourceId),
           });
+
+          if (event.ResourceType === 'AWS::CloudFormation::Stack' && !CFN_SUCCESS_STATUS.includes(event.ResourceStatus ?? '')) {
+            // If the event is not for `this` stack, recursively call for events in the nested stack
+            if (event.PhysicalResourceId !== stackToPollForEvents && event.LogicalResourceId !== stackToPollForEvents) {
```

Comment:
Why are we checking for both here? `PhysicalResourceId` and `LogicalResourceId` should never equal each other, and `stackToPollForEvents` can only ever contain one. It looks we're assigning the value of `PhysicalResourceId`, so why do we compare against `LogicalResourceId`?

Reply:
yeah I had just `PhysicalResourceId` earlier and then got confused with this. Reverted it.
https://github.com/aws/aws-cdk/blob/4e8c9c4dfdae690d9f6650bbc57bacdb83dec68c/packages/aws-cdk/lib/api/util/cloudformation/stack-activity-monitor.ts#L545-L547